.github/workflows/cleanup-cache.yml

```diff
@@ -0,0 +1,37 @@
+name: Cleanup Cache
+
+# for testing
+on: [pull_request]
+
+# on:
+#   pull_request:
+#     types:
+#       - closed
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        
+      - name: Cleanup
+        run: |
+          gh extension install actions/gh-actions-cache
+          
+          REPO=${{ github.repository }}
+          BRANCH="refs/pull/${{ github.event.pull_request.number }}/merge"
+
+          echo "Fetching list of cache key"
+          cacheKeysForPR=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
```

Comment:
I think if we remove `-L 100` it will clear the entire cache instead of only the oldest 100

Reply:
Max limit is 100: https://github.com/actions/gh-actions-cache#list

Reply:
Can we paginate and delete them all?

Reply:
Updated